### PR TITLE
[WIP] Benchmarking Streaming

### PR DESCRIPTION
**Pull request by alexeykudinkin**
## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/_raylet.pyx [hunk 1]
```diff
@@ -1262,20 +1255,32 @@ async def execute_streaming_generator_async(
...
-            context)
-        if done:
+
+        futures.append(
+            # Run it in a separate thread to that we can
+            # avoid blocking the event loop when serializing
+            # the output (which has nogil).
+            loop.run_in_executor(
+                worker.core_worker.get_thread_pool_for_async_event_loop(),
+                report_streaming_generator_output,
```
Comment (rkooo567) [inline python/ray/_raylet.pyx:1269]: I think this means report_streaming_generator_output should be thread-safe?

File: python/ray/_raylet.pyx [hunk 2]
```diff
@@ -1262,20 +1255,32 @@ async def execute_streaming_generator_async(
...
-        # avoid blocking the event loop when serializing
-        # the output (which has nogil).
-        done = await loop.run_in_executor(
-            worker.core_worker.get_thread_pool_for_async_event_loop(),
-            report_streaming_generator_output,
-            output_or_exception,
-            context)
-        if done:
+
+        futures.append(
```
Comment (rkooo567) [inline python/ray/_raylet.pyx:1263]: So IIUC, the change here is that you don't wait for each report_streaming_generator_output right?

**Comment by rkooo567** [comment]
Should I review this PR or the other one? or both?

**Comment by alexeykudinkin** [comment]
@rkooo567 ignore this one, i'm using it for experimentation. I'm putting out worthy optimizations as separate PRs and tagged you on the first batch:

 - https://github.com/ray-project/ray/pull/42259
 - https://github.com/ray-project/ray/pull/42260

**Pull request closed without merging**